src/CTA.Rules.Metrics/MetricsTransformer.cs

```diff
@@ -52,10 +52,11 @@ internal static IEnumerable<TargetVersionMetric> TransformTargetVersions(Metrics
         {
             var projectFile = projectResult.ProjectFile;
             var targetVersions = projectResult.TargetVersions;
+            var sourceVersions = projectResult.SourceVersions;
             var targetVersionMetrics = new List<TargetVersionMetric>();
-            foreach (var targetVersion in targetVersions)
+            for (int i = 0; i < targetVersions.Count; i++)
             {
-                targetVersionMetrics.Add(new TargetVersionMetric(context, targetVersion, projectFile));
+                targetVersionMetrics.Add(new TargetVersionMetric(context, targetVersions[i], projectFile, sourceVersions[i]??null));
```

Comment:
It doesn't look like we enforce `projectResult.TargetVersions` and `projectResult.SourceVersions` to have the same number of elements.

Can we have a quick check or process to reconcile when the two data structures are different lengths?

Reply:
Added a check.